### ISSUE TITLE
Constrain type parameter of `RootSpanBuilder::on_request_end` with `actix_web::body::MessageBody`

### ISSUE THREAD

**Issue by rharish101** [LukeMathWalker/tracing-actix-web#92]
Is it possible to change the API to change the following in `RootSpanBuilder`:
```rust
fn on_request_end<B>(span: Span, outcome: &Result<ServiceResponse<B>, Error>);
```
to the following?:
```rust
fn on_request_end<B: MessageBody>(span: Span, outcome: &Result<ServiceResponse<B>, Error>);
```

This will allow using methods defined for MessageBody when creating custom `RootSpanBuilder`s, such as `MessageBody::size`. The root span is always called on `MessageBody`s anyway, as seen here:
https://github.com/LukeMathWalker/tracing-actix-web/blob/c95a781204e8075ebb8b6116f280f0cfdb88dcb5/src/middleware.rs#L181-L198

I can create a PR if this is fine.

**Comment by LukeMathWalker** [comment]
Yes, a PR would be welcome!

**Comment by rharish101** [comment]
I've created PR #93 to solve this.

**Comment by LukeMathWalker** [comment]
Merged and released as 0.7.0 - thanks!
